### PR TITLE
docs(instance): Add migration v1 to v2 documentation

### PR DESCRIPTION
**Pull request by remyleone**

### CODE REVIEW

File: docs/migration_guide_v2.md [hunk 1]
```diff
@@ -0,0 +1,632 @@
...
+`scw attach` will connect to the serial port of your Scaleway instance.
+It will create a serial connection to your Scaleway server and make it available in your terminal.
+For instance, if you want to open a serial port to the `foobar` instance you would use
+
+```shell
+# v1
+scw attach foobar
+```
+
+In CLI v2, we don't offer at the moment any support for connecting to a serial port.
```
Comment (kindermoumoute) [inline docs/migration_guide_v2.md]:
```suggestion
In CLI v2, we do not offer at the moment any support for connecting to a serial port.
```

File: docs/migration_guide_v2.md [hunk 2]
```diff
@@ -0,0 +1,632 @@
...
+As a team, we don't have the bandwidth to maintain all the dependencies, perform regular audits and open-source management required to make all those dependencies thrive.
+With the CLIv2, we want to minimize them as much as possible and focus only on well-supported external libraries when required.
+
+### Few tests
+
+CLI v1 didn't have high test coverage and no test generation that could be inferred from an underlying SDK.
+CLI v2 builds on top of the tests infrastructure of the code generation we have to increase test coverage.
+We also support different types of tests: unit test, conformance test, End to End testing.
+
+## Commands
```
Comment (loicbourgois) [inline docs/migration_guide_v2.md]: The title should explain that what is listed below are `commands equivalences`, not just `commands`

File: docs/migration_guide_v2.md [hunk 3]
```diff
@@ -0,0 +1,647 @@
...
+## Justification for a v2
+
+### Multiple products design
+
+CLI v1 was created while targeting a single Scaleway Elements product: instance.
+Scaleway got many more products in its Elements ecosystem that need to be available from the CLI.
+Having ad-hoc commands for all our products was not going to be a scalable solution.
+We needed to have a more systematic approach across all products.
+
+In CLI v2, we have a common syntax across all our products: service sub-resource verb.
```
Comment (loicbourgois) [inline docs/migration_guide_v2.md]: Can accentuate on `service` `sub-resource` `verb`, with bold, italic, or single back quote ?

File: docs/migration_guide_v2.md [hunk 4]
```diff
@@ -0,0 +1,647 @@
...
+As an example:
+
+```shell
+# v2
+scw instance server list
+```
+
+* **instance**: Refers to the product namespace
+* **server**: Refers to a resource maintained in the product namespace
+* **list**: Refers to a verb apply to the currently selected API
```
Comment (loicbourgois) [inline docs/migration_guide_v2.md]: apply -> applied

File: docs/migration_guide_v2.md [hunk 5]
```diff
@@ -0,0 +1,647 @@
...
+
+In English, it would be: "list all servers available on the instance API".
```
Comment (loicbourgois) [inline docs/migration_guide_v2.md]: In English -> In plain English

File: docs/migration_guide_v2.md [hunk 6]
```diff
@@ -0,0 +1,647 @@
...
+
+### Consistent workflow across products
+
+CLI v1 was designed to offer a syntax close to the Docker syntax.
+For instance, running a command such as `echo foobar` on a remote server is `scw run ubuntu-bionic echo foobar` which mimic `docker run ubuntu echo foobar`.
```
Comment (loicbourgois) [inline docs/migration_guide_v2.md]: to run a command such as `..`, we can write `...`

File: docs/migration_guide_v2.md [hunk 7]
```diff
@@ -0,0 +1,647 @@
...
+While this can be useful for some tasks, there are plenty of actions that do not fit in with this paradigm.
+E.g., attaching to a running server sub-resources such as volumes or security groups are not performed easily using this paradigm because Docker doesn't provide the same features.
+
+In CLI v2 we offer support for a wide range of actions on all resources present and coming in the Scaleway Elements ecosystem.
+Actions are organized around a set of verbs such as `list`, `get`, `create`, `update` that can be used with a wide variety of products and do not suppose any preconceived workflow.
+
+### Automated code generation
+
+CLI v1 did not have any code generation features to easily create supports in all SDKs and developer tools we support.
```
Comment (loicbourgois) [inline docs/migration_guide_v2.md]: It's not the CLI that has code generation feature. 

It's code generation that is used to build the CLI, and also the SDKs and other dev tools. 

`CLI v1 did not take advantage of code generation features to easily create supports like in SDKs and other developper tools`

File: docs/migration_guide_v2.md [hunk 8]
```diff
@@ -0,0 +1,647 @@
...
+
+* [scaleway-cli](https://github.com/scaleway/scaleway-cli)
+* [scaleway-sdk-go](https://github.com/scaleway/scaleway-sdk-go)
+* [scaleway provider for Terraform](https://github.com/terraform-providers/terraform-provider-scaleway/)
+
+### Cleanup of old and unmaintained dependencies
+
+CLI v1 required a lot of dependencies that are not actively maintained anymore.
+
+With the CLIv2, we want to minimize them as much as possible and focus only on well-supported external libraries when required.
```
Comment (loicbourgois) [inline docs/migration_guide_v2.md]: CLIv2 -> CLI v2

File: docs/migration_guide_v2.md [hunk 9]
```diff
@@ -0,0 +1,647 @@
...
+* **instance**: Refers to the product *namespace*
+* **server**: Refers to a *resource* maintained in the product namespace
+* **list**: Refers to a *verb* applied to the currently selected API
+
+In plain English, it would be: "list all servers available on the instance API".
+
+### Consistent workflow across products
+
+CLI v1 was designed to offer a syntax close to the Docker syntax.
+For instance, to run a command such as `echo foobar` on a remote server, we can write `scw run ubuntu-bionic echo foobar` which mimic `docker run ubuntu echo foobar`.
```
Comment (loicbourgois) [inline docs/migration_guide_v2.md]:
```suggestion
For instance, to run a command such as `echo foobar` on a remote server, we can write `scw run ubuntu-bionic echo foobar` which mimics `docker run ubuntu echo foobar`.
```

File: docs/migration_guide_v2.md [hunk 10]
```diff
@@ -0,0 +1,647 @@
...
+```shell
+# v1
+scw events
+```
+
+You can still access those data using the API:
+
+```shell
+# On fr-par-1
+curl -H "X-Auth-Token: $SCW_SECRET_KEY" https://cp-par1.scaleway.com/tasks
```
Comment (kindermoumoute) [inline docs/migration_guide_v2.md]: write down actual urls `api.scaleway.com/instance/v1/zones/fr-par-1/...`
